nf_core/components/nfcore_component.py

```diff
@@ -279,3 +279,23 @@ def get_outputs_from_main_nf(self):
                     pass
             log.debug(f"Found {len(outputs)} outputs in {self.main_nf}")
             self.outputs = outputs
+
+    def get_ext_args_from_main_nf(self) -> List[str]:
+        """Collect all ext.args from the main.nf file."""
+        extargs: set[str] = set()
+        with open(self.main_nf) as f:
+            data = f.read()
+        if self.component_type == "modules":
```

Comment:
in theory this should only be run with modules, so I think we can either avoid this if, or run everything under the if

Reply:
I have moved the code under he `if` but I still choose to return an empty list for the implicit else as this makes it easier for type-checking and means we do not require an `if not None` type-gate.